crates/sui-node/src/lib.rs

```diff
@@ -1240,6 +1240,15 @@ impl SuiNode {
                 }
             };
             *self.validator_components.lock().await = new_validator_components;
+
+            #[cfg(msim)]
+            self.state
+                .prune_checkpoints_for_eligible_epochs(
```

Comment:
It might make sense to also invoke compaction here under `msim` to truly delete txns and effects. Given we use range delete based tombstones and ignore them during reads, there is a good chance tx/effects haven't been purged from db yet after this call.

Reply:
Oh good callout, thanks will add that